gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/task/DagProcessingEngineMetrics.java

```diff
@@ -0,0 +1,177 @@
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+import java.util.concurrent.ConcurrentMap;
+import javax.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.gobblin.configuration.State;
+import org.apache.gobblin.instrumented.Instrumented;
+import org.apache.gobblin.metrics.ContextAwareMeter;
+import org.apache.gobblin.metrics.GobblinMetrics;
+import org.apache.gobblin.metrics.MetricContext;
+import org.apache.gobblin.metrics.MetricTagNames;
+import org.apache.gobblin.metrics.Tag;
+import org.apache.gobblin.service.modules.orchestration.DagActionStore;
+import org.apache.gobblin.metrics.ServiceMetricNames;
+
+/**
+ * Used to track all metrics relating to processing dagActions when DagProcessingEngine is enabled. The metrics can be
+ * used to trace the number of dagActions (which can be further broken down by type) at various points of the system,
+ * starting from addition to the DagActionStore to observation in the DagActionChangeMonitor through the
+ * DagProcessingEngine pipeline (DagManagement -> DagTaskStreamImpl -> MySqlMultiActiveLeaseArbiter -> DagProc).
+ */
+@Slf4j
+public class DagProcessingEngineMetrics {
+  MetricContext metricContext;
+  // Declare map of dagActionType to a ContextAwareMeter for each metric
+  private ConcurrentMap<String, ContextAwareMeter> dagActionsStoredMeterByDagActionType = new ConcurrentHashMap();
+  private ConcurrentMap<String, ContextAwareMeter> dagActionsObservedMeterByDagActionType =  new ConcurrentHashMap();
+  private ConcurrentMap<String, ContextAwareMeter> dagActionsLeasesObtainedMeterByDagActionType =  new ConcurrentHashMap();
+  private ConcurrentMap<String, ContextAwareMeter> dagActionsNoLongerLeasingMeterByDagActionType =  new ConcurrentHashMap();
+  private ConcurrentMap<String, ContextAwareMeter> dagActionsLeaseReminderScheduledMeterByDagActionType =  new ConcurrentHashMap();
+  private ConcurrentMap<String, ContextAwareMeter> dagActionsReminderProcessedMeterByDagActionType =  new ConcurrentHashMap();
+  private ConcurrentMap<String, ContextAwareMeter> dagActionsExceededMaxRetryMeterByDagActionType =  new ConcurrentHashMap();
+  private ConcurrentMap<String, ContextAwareMeter> dagActionsInitializeFailedMeterByDagActionType =  new ConcurrentHashMap();
+  private ConcurrentMap<String, ContextAwareMeter> dagActionsInitializeSucceededMeterByDagActionType =  new ConcurrentHashMap();
+  private ConcurrentMap<String, ContextAwareMeter> dagActionsActFailedMeterByDagActionType =  new ConcurrentHashMap();
+  private ConcurrentMap<String, ContextAwareMeter> dagActionsActSucceededMeterByDagActionType =  new ConcurrentHashMap();
+  private ConcurrentMap<String, ContextAwareMeter> dagActionsConcludeFailedMeterByDagActionType =  new ConcurrentHashMap();
+  private ConcurrentMap<String, ContextAwareMeter> dagActionsConcludeSucceededMeterByDagActionType =  new ConcurrentHashMap();
```

Comment:
why should these not be of type `ConcurrentMap<DagActionStore.DagActionType, ContextAwareMeter>` ?
that will avoid unnecessary calls to `.toStrings()` later